specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/1.0.0/iotcentral.json

```diff
@@ -0,0 +1,2262 @@
+          }
+        ],
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceCommandCollection"
+            }
+          }
+        }
+      },
+      "post": {
+        "operationId": "Devices_ExecuteComponentCommand",
+        "summary": "Execute a component command",
+        "description": "Execute a command on a component.",
+        "x-ms-examples": {
+          "Send command": {
+            "$ref": "./examples/devices_commands_send.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "$ref": "#/parameters/CommandName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device command body.",
+            "schema": {
+              "$ref": "#/definitions/DeviceCommand"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceCommand"
+            }
+          }
+        }
+      }
+    },
+    "/devices/{device_id}/components/{component_name}/properties": {
+      "get": {
+        "operationId": "Devices_GetComponentProperties",
+        "summary": "Get device properties for a specific component",
+        "x-ms-examples": {
+          "Get properties on components": {
+            "$ref": "./examples/devices_properties_components_get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_ReplaceComponentProperties",
+        "summary": "Replace device properties for a specific component",
+        "x-ms-examples": {
+          "Update properties on components": {
+            "$ref": "./examples/devices_properties_components_set.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "$ref": "#/parameters/ComponentName"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device properties.",
+            "schema": {
+              "$ref": "#/definitions/DeviceProperties"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "202": {
```

Comment:
Why 202? Is it long running?

Reply:
PUT properties isn't creating anything, but it's also not actually handled synchronously, thus 202 (Accepted) rather than 201 (Created) or 200 (OK)